crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -192,8 +194,10 @@ pub fn extract_meshes(
     let mut not_caster_commands = Vec::with_capacity(*prev_not_caster_commands_len);
     let visible_meshes = meshes_query.iter().filter(|(_, vis, ..)| vis.is_visible());
 
-    for (entity, _, transform, previous_transform, handle, not_receiver, not_caster) in
-        visible_meshes
+    let mut visible_mesh_entities = vec![Entity::PLACEHOLDER];
```

Comment:
Instead of having `VisibleMeshEntities` in the render world, and pass it to the main world for each picking camera, what about just updating a `VisibleMeshEntities` in the Main world?

It's a big deal, given this might be a pretty large array and it's allocated & cloned every frame (currently).

Also keeping it in the main world will not prevent you from updating it from this system (so that you still have correct ordering to match the `mesh_id`)

Reply:
Good question, I don't remember why I did it this way 😅. I'll look into doin that instead

Reply:
Did you look into it?

Reply:
Not yet. I started looking into it a bit, but hit a blocker and stopped. I'm planning on trying again next week.